fth/t_tools.fth

```diff
@@ -72,11 +84,18 @@ CREATE the-test 128 CHARS ALLOT
         DEPTH 0
         ?DO             \ for each stack item
             actual-results I CELLS + @ \ compare actual with expected
-            <>
-            IF
+            2dup
+            =
+            if
+                2drop
+            else
+                >R >R
                 -1 test-passed +!
                 1 test-failed +!
-                S" INCORRECT RESULT: " error
+                s" INCORRECT RESULT (expected=" R> int>str concat
+                s" , got="         concat+free  R> int>str concat+free
```

Comment:
There may be a memory leak here because ERROR does not free the memory allocated by CONCAT.
Rather that trying to insert the new message into the ERROR message, let's just add this after the call to ERROR.

```suggestion
               ."  (expected=" r> .
               ." , got=" r> .
               ." )"
```

Then we don't need the INT>STR or CONCAT tools.

Reply:
Hmm, having the values at the start of each line makes it easier to read for me.

Could we perhaps directly output the values **BEFORE** calling error?